realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionChangeListener.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+/**
+ * {@link OrderedRealmCollectionChangeListener} can be registered with a {@link RealmResults} to receive a notification
+ * with a {@link OrderedCollectionChange} to describe the details of what have been changed in the collection from last
+ * time.
+ * <p>
+ * Realm instances on a thread without an {@link android.os.Looper} cannot register a
+ * {@link OrderedRealmCollectionChangeListener}.
+ * <p>
+ *
+ * @see {@link RealmResults#addChangeListener(OrderedRealmCollectionChangeListener)}.
+ */
+public interface OrderedRealmCollectionChangeListener<T> {
+    /**
+     * This will be called when the async query finished at the first time or the collection gets changed.
+     *
+     * @param collection the collection this listener is registered to.
+     * @param changes {@code null} if the it is first time async query returns or information about which rows in the
```

Comment:
@realm/java the `changes` will be `null` if it is the first time async query returns. It is the same behaviour like cocoa. Anyone think it should return an empty `OrderedCollectionChange` instead? in that case, we need a new method `OrderedCollectionChange.isEmpty()` which feels redundant.